### ISSUE TITLE
Plugin crashes when creating an oauth client using user token instead of client id, secret.

### ISSUE THREAD

**Issue by msd86911** [MyPureCloud/terraform-provider-genesyscloud#1301]
When creating a genesyscloud_oauth_client using user access token (instead of client Id, secret), it is failing with error 
_**invalid memory address or nil pointer dereference**_

Here is the error trace from debug:
`goroutine 487 [running]:
terraform-provider-genesyscloud/genesyscloud/oauth_client.updateTerraformUserWithRole({0x2b1f8e8, 0xc0007dd530}, 0xc00080cea0, 0x0)
	D:/Workspace/Extra/Terraform/TerraformCode/terraform-provider-genesyscloud/genesyscloud/oauth_client/resource_genesyscloud_oauth_client.go:212 +0x7ea
terraform-provider-genesyscloud/genesyscloud/oauth_client.createOAuthClient({0x2b1f8e8, 0xc0007dd530}, 0xc000a6da00, {0x25174a0, 0xc0007dd590})
	D:/Workspace/Extra/Terraform/TerraformCode/terraform-provider-genesyscloud/genesyscloud/oauth_client/resource_genesyscloud_oauth_client.go:61 +0x4cc
`
I am trying to create two clients, the first one is Client credential and second is Code authorization.
I tried divide them and add wait time in between, now I see the first client is created and second one is failing.
The error in line 212 (function updateTerraformUserWithRole), the app is crashing in the following for loop.
//Step #4  - Concat the addedRoles
```
for _, role := range *addedRoles {
	totalRoles = append(totalRoles, *role.RoleId)
}
```

**Comment by msd86911** [comment]
I found the issue, I just checked the addRoles before going into the for loop and now it works.
```
//Step #4  - Concat the addedRoles
**if addedRoles != nil {**
	for _, role := range *addedRoles {
		totalRoles = append(totalRoles, *role.RoleId)
	}
**}**
```
Can you please add it into your next release?

**Comment by charliecon** [comment]
Hi @msd86911  

Sorry for the delayed response. We are doing our release today and we will try to include this fix. 

Thanks for reporting.
-Charlie

**Comment by charliecon** [comment]
Fixed as of v1.50.0 ✅
